elastic/elastic-agent
# Refactor pkg/testing and small improvements to testing/integration

### commit title: `refactor testing/integration and pkg/testing`
### commit body:
```
* pkg/testing/tools
  - split the 'tools' into smaller packages with well defined scope: 'estools', 'fleettools' and 'check'
  - adjust method names for consistency

* testing/integration:
  - make diagnostics to require to be run on a remote host
  - add more context to errors
  - reorder functions on endpoint_security_test so tests are the first functions and when the test fails because the 'Endpoint' directory isn't removed, it now prints the content of the directory
```

## What does this PR do?

Refactor pkg/testing and small improvements to testing/integration

* pkg/testing/tools
  - split the 'tools' into smaller packages with well defined scope: 'estools', 'fleettools' and 'check'
  - adjust method names for consistency

* testing/integration:
  - make diagnostics to require to be run on a remote host
  - add more context to errors
  - reorder functions on endpoint_security_test so tests are the first functions and when the test fails because the 'Endpoint' directory isn't removed, it now prints the content of the directory


## Why is it important?

More context to errors make it easier to understand why a test failed. The refactoring helps the discoverability of the existing tools/helpers as well as the overal code readability and maintainability. 

## Checklist





## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
